aws-cli/bash-linux/s3/bucket-lifecycle-operations/general.sh

```diff
@@ -0,0 +1,100 @@
+# No 'shebang' (#!) because this script is intended to 'source', not run.
+###############################################################################
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# This file is licensed under the Apache License, Version 2.0 (the "License").
+#
+# You may not use this file except in compliance with the License. A copy of
+# the License is located at http://aws.amazon.com/apache2.0/.
+#
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+###############################################################################
+
+# This script contains general purpose functions that are used throughout 
+# the AWS Command Line Interface (AWS CLI) code samples that are maintained
+# in the repo at https://github.com/awsdocs/aws-doc-sdk-examples
+#
+# They are intended to abstract functionality that is required for the tests
+# to work without cluttering up the code. The intent is to ensure the purpose
+# of the code is clear.
+
+
+
+###############################################################################
+# function TESTS-FAILED
+#
+# This function is used to terminate a failed test and to warn the customer
+# about possible undeleted resources that could incur costs to their account.
+###############################################################################
+
+function TESTS-FAILED {
+    echo ""
+    echo "TESTS FAILED"
+    echo "One or more of the tests failed to complete successfully. This means"
+    echo "that any tests that were supposed to run after the failed test did'nt"
```

Comment:
Edit: "... the failed test didn't..." (there's a typo)

Reply:
Fixed.